test/jobs/image_loader_job_test.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+# Mock classes to simulate google cloud storage objects.
+class MockStorage
+  def initialize
+    @buckets = {}
+  end
+
+  def bucket(name)
+    @buckets[name] ||= MockBucket.new
+  end
+end
+
+class MockBucket
+  def initialize
+    @files = {}
+  end
+
+  def file(path)
+    @files[path] ||= MockFile.new
+  end
+end
+
+class MockFile
+  def download(path)
+    # Simulate file download by creating a test file
+    FileUtils.mkdir_p(File.dirname(path))
+    File.write(path, "mock file content")
+  end
+end
+
+class ImageLoaderJobTest < ActiveJob::TestCase
+  DIR = Rails.root.join("tmp/downloads")
+
+  def setup
+    @mock_storage = MockStorage.new
+    @test_image = Image.first
+    @test_file = "#{DIR}/#{@test_image.id}.jpg"
+    FileUtils.rm_rf(DIR)
+  end
+
+  def with_stubs(&block)
+    Google::Cloud::Storage.stub(:new, @mock_storage) do
+      MO.stub(:local_original_image_cache_path, DIR, &block)
+    end
+  end
+
+  def teardown
+    FileUtils.rm_rf(DIR)
+  end
+
+  test "image download works" do
+    rolf_count = users(:rolf).original_image_quota
+    admin_count = User.admin.original_image_quota
+    with_stubs do
+      ImageLoaderJob.perform_now(@test_image.id, users(:rolf).id)
```

Comment:
Question: I thought tests always perform_now. Is that correct?

Reply:
Ah, this is something I actually know something about!  No.  It was essentially random before, but now I have it configured

```
config.active_job.queue_adapter = :test
```

Which means the jobs do not run unless you explicitly queue them as "perform_now".

It may be the case that they were set to start immediately before, but if so, they were running asynchronously resulting in random behavior in the tests that really makes testing a nightmare.  It was acting like the jobs were persisting from one test to another in some cases, which is just crazy because of stubs changing.  I have no idea what the expected behavior in such a case would be!!

I use `ImageLoaderJob.perform_now` to test the `perform` method in test/jobs/image_loader_job.test.rb, and that seems to be stable.